### PR TITLE
Refactor git_clone_foreman_infra function to not require args map

### PR DESCRIPTION
**Pull request by zjhuntin**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy [hunk 1]
```diff
@@ -1,6 +1,4 @@
-void git_clone_foreman_infra(args) {
-    target_dir = args.target_dir ?: ''
-
+void git_clone_foreman_infra(target_dir = '') {
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy]: This signature shouldn't need changing.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/duffy.groovy [hunk 2]
```diff
@@ -1,6 +1,6 @@
 def provision() {
     fix_ansible_config()
-    git_clone_foreman_infra(dir: 'foreman-infra')
+    git_clone_foreman_infra(target_dir = 'foreman-infra')
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/duffy.groovy]: I think this should just be `target_dir:`

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy (hunk 1 above)
Comment (zjhuntin) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy]: @ehelms  [here](https://github.com/theforeman/foreman-infra/blob/master/puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foremanRelease.groovy#L33) it is failing at this point at the moment. It doesn't seem to recognize that args.target_dir should exist and fails out. Is there another way to not have it do that?

[This](http://ci.theforeman.org/job/foreman-nightly-release/326/) is an example failure

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy]: Oh maybe we need to set a default value for args, something like `args = {}` in the parameters.

**Comment by tbrisker** [comment]
Is it possible this didn't propagate to centos ci yet? I'm still seeing nightly failures with the same error.

**Comment by ekohl** [comment]
So the deploy job is failing to deploy its own fix. Nice!

**Pull request merged**